data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -326,6 +348,25 @@ public void doOutput(final Collection<Record<Event>> records) {
       lastFlushTimeMap.put(threadId, System.currentTimeMillis());
     }
 
+    if (responseLatencyAlertRatio.get() > 1.0) {
```

Comment:
This class has been growing and getting lots of responsibilities. We can refactor this such that the `OpenSearchSink` class is mostly immune to these changes.

There is one public method on `BulkRetryStrategy` - `void execute(final AccumulatingBulkRequest bulkRequest)`. We can create an interface for this method.

```
interface BulkRequestExecutor {
  void execute(final AccumulatingBulkRequest bulkRequest)
}
```

Then make `BulkRetryRequest` implement this. 

Now, we can make a `DelayRequestExecutor` which has the logic for the backoff. It would effectively have this code.

```
class DelayRequestExecutor implements BulkRequestExecutor {
  private final BulkRequestExecutor innerExecutor;

  void execute(final AccumulatingBulkRequest bulkRequest) {
    // backoff code
    innerExecutor.execute(bulkRequest);
  }
```